src/main/java/seedu/address/model/student/Student.java

```diff
@@ -52,6 +57,46 @@ public Student(Name name, Phone phone, Email email, AcademicYear academicYear, S
         this.tags.addAll(tags);
     }
 
+    /**
+     * Adds a training session to the student's schedule.
+     * Training sessions are automatically sorted by their respective date and times.
+     *
+     * @param trainingDateTime LocalDateTime corresponding to the training's date and start time.
+     * Duplicates are not allowed and will not be added.
+     */
+    public void addTraining(LocalDateTime trainingDateTime) {
+        trainingSchedules.add(trainingDateTime);
+    }
+
+    /**
+     * Checks if student has a training scheduled at the specified date and start time.
+     *
+     * @param trainingDateTime LocalDateTime corresponding to the training's date and start time.
+     * Specified training must exist in the student's training schedule.
+     * @return true If a training has been scheduled at the specified date and time.
+     */
+    public boolean containsTraining(LocalDateTime trainingDateTime) {
+        return trainingSchedules.contains(trainingDateTime);
+    }
+
+    /**
+     * Removes a scheduled training from the student's training schedule.
+     *
+     * @param trainingDateTime LocalDateTime corresponding to the training's date and start time.
+     */
+    public void removeTraining(LocalDateTime trainingDateTime) {
+        if (containsTraining(trainingDateTime)) {
+            trainingSchedules.remove(trainingDateTime);
+        }
+    }
+
+    /**
+     * Clears all trainings scheduled for the student.
+     */
+    public void clearAllTraining() {
```

Comment:
I personally think `removeAllTraining()` or `clearTraining()` would be more coherent with the removeTraining method above, in terms of naming. It's more of an opinion only so you still make the final decision

Reply:
Changed to removeAllTraining()